### DISCUSSION TITLE
How to get callback when device is connected to sink

### DISCUSSION THREAD

**Discussion by daywink** [pschatzmann/ESP32-A2DP#334] [Q&A] [answered]
Hello, I am trying to make a simple Bluetooth speaker with a display, and I want to change the text from "Connecting..." to "Connected!" when a device is connected without using a looping if statement. Any help would be nice

Sorry if this is the wrong place to post, I don't know where else to ask.

**Comment by pschatzmann** [comment, marked as answer]
This is definitely not an issue, so I convert it to a discussion. I suggest that you have a look at the class documentation: There you find the set_on_connection_state_changed callback registration method. There is even an example in the examples directory!
